packages/web3/src/signer/signer.ts

```diff
@@ -161,11 +67,23 @@ export interface SignerProvider {
   signMessage(params: SignMessageParams): Promise<SignMessageResult>
 }
 
+// Abstraction for interactive signer (e.g. WalletConnect instance, Extension wallet object)
+export interface InteractiveSignerProvider<ConnectOptions extends { chainGroup?: number }> extends SignerProvider {
+  // @param chainGroup - specify whether to use addresses from a specific group
+  connect(opt?: ConnectOptions): Promise<void>
```

Comment:
Should it be renamed as `enable` as `Argent-X`?

Reply:
Sure :+1:

Reply:
What should we rename 'ConnectOptions'? `EnableOptions` sounds good?

Reply:
Done!